com.unity.netcode.adapter.utp/Runtime/UTPTransport.cs

```diff
@@ -0,0 +1,836 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+using NGONetworkEvent = Unity.Netcode.NetworkEvent;
+using UTPNetworkEvent = Unity.Networking.Transport.NetworkEvent;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Collections;
+using Unity.Networking.Transport;
+using Unity.Networking.Transport.Relay;
+using Unity.Networking.Transport.Utilities;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Provides an interface that overrides the ability to create your own drivers and pipelines
+    /// </summary>
+    public interface INetworkStreamDriverConstructor
+    {
+        void CreateDriver(UTPTransport transport, out NetworkDriver driver, out NetworkPipeline unreliableSequencedPipeline, out NetworkPipeline reliableSequencedPipeline, out NetworkPipeline reliableSequencedFragmentedPipeline);
+    }
+
+    public class UTPTransport : NetworkTransport, INetworkStreamDriverConstructor
+    {
+        public enum ProtocolType
+        {
+            UnityTransport,
+            RelayUnityTransport,
+        }
+
+        private enum State
+        {
+            Disconnected,
+            Listening,
+            Connected,
+        }
+
+        public const int MaximumMessageLength = 6 * 1024;
```

Comment:
where this number is coming from?

Reply:
Boss Room :( It is still configurable in the serialized field.